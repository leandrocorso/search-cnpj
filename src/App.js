import React from 'react';
import Routes from './routes';
import ThemeProvider from 'styled-components';
import theme from './theme';

const App = () => (
    <Routes>
        <ThemeProvider theme={theme} />
    </Routes>
);

export default App;
