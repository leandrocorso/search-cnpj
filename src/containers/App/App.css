@import url('https://fonts.googleapis.com/css?family=Roboto');

* {
    margin: 0;
    padding: 0;
    border: 0;
    box-sizing: border-box;
    outline: none;
}

html, body {
    height: 100%;
    width: 100%;
}

body {
    font-family: 'Roboto', sans-serif;
    font-size: '.9rem';
    color: #333;
    background-color: #fff;
}

#root {
    max-width: 420px;
    height: 100vh;
    margin: 0 auto;
    background-color: #eee;
}

div.container {
    padding: 1em;
}