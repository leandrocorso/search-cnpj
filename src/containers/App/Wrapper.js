import React, { Component } from 'react';

import { Auth } from '../../utils';

class Wrapper extends Component {

	componentDidMount() {
		Auth.setToken();
	}

	render() {
		return (
			<React.Fragment>
				{this.props.children}
			</React.Fragment>
		);
	}
}

export default Wrapper;
