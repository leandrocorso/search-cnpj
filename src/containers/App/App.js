import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import { GlobalStyle } from '../../styles';

import Wrapper from './Wrapper';
import Home from '../Home';
import SearchCnpj from '../SearchCnpj';

const App = () => (
    <BrowserRouter>
        <Wrapper>
            <GlobalStyle />
            <Switch>
                <Route className="testHome" exact path="/" component={Home} />
                <Route className="testSearch" exact path="/busca-cnpj" component={SearchCnpj} />
            </Switch>
        </Wrapper>
    </BrowserRouter>
);

export default App;
