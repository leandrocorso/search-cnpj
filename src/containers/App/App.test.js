import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('Testing App component', () => {
    
    const wrapper = shallow(<App />);
    
    test('App exists', () => {
        expect(wrapper);
    });

    test('App have the browser router', () => {
        expect(wrapper.find('BrowserRouter'));
    });

    test('App have the global style component', () => {
        expect(wrapper.find('GlobalStyleComponent'));
    });

    test('App have the path to home', () => {
        expect(wrapper.find('.testHome')).toHaveProp({ path : '/'});
    });

    test('App have the path to search page', () => {
        expect(wrapper.find('.testSearch')).toHaveProp({ path : '/busca-cnpj'});
    });

    test('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})