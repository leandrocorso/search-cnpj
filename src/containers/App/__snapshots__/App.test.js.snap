// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component Should render correctly 1`] = `
<BrowserRouter>
  <Wrapper>
    <GlobalStyleComponent />
    <Switch>
      <Route
        className="testHome"
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        className="testSearch"
        component={[Function]}
        exact={true}
        path="/busca-cnpj"
      />
    </Switch>
  </Wrapper>
</BrowserRouter>
`;
